sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,136 @@
+}
+
+static void execute_test(dif_aon_timer_t *aon_timer, uint32_t sleepTime_us) {
+  time_elapsed = 0;
+  peripheral = kTopEarlgreyPlicPeripheralUnknown;
+  irq = kDifAonTimerIrqWdogTimerBark;
+
+  // Setup the timer and wait the IRQ
+  uint32_t sleep_cycles = (sleepTime_us * kAonTimerFreq / 1000000);
+  LOG_INFO("Waiting wakeup interrupt for %u/%u (cycles/us)", sleep_cycles,
+           sleepTime_us);
+  // Capture the current tick to measure the time the IRQ will take
+  timeout = ibex_timeout_init(sleepTime_us * 1000);
+  aon_timer_wakeup_config(aon_timer, sleep_cycles);
+  wait_for_interrupt();
+
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralAonTimerAon,
+        "Interrupt from incorrect peripheral: exp = %d, obs = %d",
+        kTopEarlgreyPlicPeripheralAonTimerAon, peripheral);
+
+  CHECK(irq == kDifAonTimerIrqWkupTimerExpired,
+        "Interrupt type incorrect: exp = %d, obs = %d",
+        kDifAonTimerIrqWkupTimerExpired, irq);
+
+  CHECK((uint32_t)time_elapsed >= sleepTime_us,
+        "Timer took %u usec which is less than %u", (uint32_t)time_elapsed,
+        sleepTime_us);
+}
+
+void ottf_external_isr(void) {
+  // Calc the elapsed time since the activation of the IRQ
+  time_elapsed = ibex_timeout_elapsed(&timeout);
+
+  dif_rv_plic_irq_id_t plic_irq_id;
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_claim(&plic, kTopEarlgreyPlicTargetIbex0, &plic_irq_id));
+
+  irq = (dif_aon_timer_irq_t)(
+      plic_irq_id -
+      (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+  peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(&aon_timer, irq));
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  // Initialize pwrmgr
+  dif_pwrmgr_t pwrmgr;
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+
+  // Initialize aon timer
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge_all(&aon_timer));
+  CHECK_DIF_OK(dif_pwrmgr_irq_acknowledge_all(&pwrmgr));
+
+  dif_pwrmgr_irq_state_snapshot_t pwrmgr_irqs =
+      (dif_pwrmgr_irq_state_snapshot_t)UINT_MAX;
+
+  CHECK_DIF_OK(dif_pwrmgr_irq_restore_all(&pwrmgr, &pwrmgr_irqs));
+
+  // Initialize the PLIC.
+  mmio_region_t plic_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(plic_base_addr, &plic));
+
+  // Enable interrupts
+  rv_plic_testutils_irq_range_enable(
+      &plic, kTopEarlgreyPlicTargetIbex0, kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
+
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge_all(&aon_timer));
+
+  execute_test(&aon_timer, kSleepTime_us);
+  execute_test(&aon_timer, 2 * kSleepTime_us);
```

Comment:
Could you please document why there is a need for two tests with a different sleep time, and what calling this test twice is trying to solve?